File: Composite Numbers/composite_numbers.py
import math

def isPrime(n):
    if n <= 3:
        return n > 1
    if n % 2 == 0 or n % 3 == 0:
        return False
    i = 5
    while i * i < n:
        if n % (i) == 0 or n % (i+2) == 0:
            return False
        i += 6
    return True
    
def main():
    n = int(input('Enter n: '))
    count = -1
    curr = 2
    while count < n:
        curr += 1
        if not isPrime(curr):
            count += 1
    print(curr)

main()

File: Composite Numbers/composite_numbers.c
#include <stdio.h>

int isPrime(int num){
    if (num <= 3)
    {
        return num > 1;
    }
    if ((num % 2 == 0) || (num % 3 == 0))
    {
        return 0;
    }
    int i = 5;
    while (i * i < num)
    {
        if ((num % i == 0) || (num % (i + 2) == 0))
        {
            return 0;
        }
        i = i + 6;
    }
    return 1;
}

int main(){
    int input;
    printf("Enter n: ");
    scanf("%d", &input);
    int count = -1;
    int curr = 2;
    while(count < input)
    {
        curr++;
        if (!isPrime(curr))
        {
            count++;
        }
    }
    printf("%d\n", curr);
}
